### PR TITLE
add lint check for pr pipeline

### PR DESCRIPTION
**Pull request by umsu2**
add lint:check and unit test as part of the pipeline for PRs.

### CODE REVIEW

File: .github/workflows/pr.yml
Comment (SamHuo213) [inline .github/workflows/pr.yml:3]: What do you think about changing this to on: pull_request?

On push seems to aggressive and will over utilize resources.

File: .github/workflows/pr.yml
Comment (SamHuo213) [inline .github/workflows/pr.yml:17]: I like the idea of running tests.

I think we should do one of the following here.

- Remove the comment
- Add test to this workflow

Comment (umsu2) [reply .github/workflows/pr.yml:17]: waiting for the test branch to be merged so this could be uncommented

Comment (umsu2) [reply .github/workflows/pr.yml]: @SamHuo213 we should enable github action require branch to be up to date to master, i didn't even realize my branch is not up to date with main

Comment (umsu2) [reply .github/workflows/pr.yml]: also after this gets merged we can enable branch protection as such that if the `buidl` job fails it won't let pr get merged.

**Review by SamHuo213 — APPROVED** [review]

File: .github/workflows/pr.yml
Comment (SamHuo213) [reply .github/workflows/pr.yml]: I like both ideas, please add tasks to our channel so someone can pick it up.

Comment (SamHuo213) [reply .github/workflows/pr.yml]: I know Github PRs can display when checks don't pass as a comment. That should be part of this change.

**Pull request merged**
